web/tests/input_pill.test.js

```diff
@@ -553,6 +556,90 @@ run_test("exit button on pill", ({mock_template}) => {
     assert.deepEqual(widget.items(), [items.red]);
 });
 
+run_test("open user popover on user's pill click", () => {
+    const info = set_up();
+    const config = info.config;
+    const $container = info.$container;
+    input_pill.create(config);
+
+    let is_user_popover_open = false;
+
+    const pill_stub = {
+        to_$: () => ({
+            attr(attribute) {
+                assert.equal(attribute, "data-user-id");
+                return 7;
+            },
+        }),
+    };
+
+    const e = {
+        stopPropagation: noop,
+    };
+
+    people.get_by_user_id = (user_id) => {
+        assert.equal(user_id, 7);
+
+        return {
+            email: "mary@example.com",
+            user_id: 7,
+            full_name: "Mary",
+        };
+    };
+
+    popovers.show_user_info_popover = (_, user) => {
+        assert.equal(user.email, "mary@example.com");
+        assert.equal(user.user_id, 7);
+        assert.equal(user.full_name, "Mary");
+        is_user_popover_open = true;
+    };
+
+    const pill_click_handler = $container.get_on_handler("click", ".pill");
+
+    assert.equal(is_user_popover_open, false);
+    pill_click_handler.call(pill_stub, e);
+    assert.equal(is_user_popover_open, true);
+});
+
+run_test("should not open popover on clicking a non-user pill", () => {
+    const info = set_up();
+    const config = info.config;
+    const $container = info.$container;
+    input_pill.create(config);
+
+    let is_user_popover_open = false;
+
+    const pill_stub = {
+        to_$: () => ({
+            attr(attribute) {
+                assert.equal(attribute, "data-user-id");
+                return null;
+            },
+        }),
+    };
+
+    const e = {
+        stopPropagation: noop,
+    };
+
+    people.get_by_user_id = () => {
+        /* istanbul ignore next */
+        assert.fail("popovers.get_by_user_id should not be called for a non-user pill");
```

Comment:
There is no need for this, I think. It is enough to test that the popover is not opened.

Reply:
I don't think we do such checks in other tests, so I don't have any strong opinions. And if someone changes the code to not return early by mistake, then the below code to check `is_user_popover_open` value will anyways fail. We can wait for feedback from others.

Reply:
Yeah. That's true. I didn't see them anywhere else so I was a bit hesitant. But I still think this is a not a wrong way to test.
> And if someone changes the code to not return early by mistake, then the below code to check is_user_popover_open value will anyways fail.

I think this should be the right behaviour. Anyone modifying the early return code should be conscious of the impact.

Anyway I agree, this may not be the "zulip" convention of doing things.